### ISSUE TITLE
Border issue with the pen command "fill" ?

### ISSUE THREAD

**Issue by nathalierun** [jmoenig/Snap--Build-Your-Own-Blocks#1537]
Hello,

Here are some images that I obtain with the fill command :

Step 3
![sierpinski aleatoire niveau3 p 0 2](https://cloud.githubusercontent.com/assets/17542520/20711250/291d61b6-b658-11e6-90e5-f523511cfedb.png)

Step 5
![sierpinski aleatoire niveau5 p 0 2](https://cloud.githubusercontent.com/assets/17542520/20711244/209c35bc-b658-11e6-92fb-a98d3e0c8ec6.png)

Step 6
![sierpinski random step6](https://cloud.githubusercontent.com/assets/17542520/20711221/06ae4d98-b658-11e6-8972-583458dd285e.png)

The filling of the triangle is perfect in step 3 anf step 4.
Step 5 : We begin to see problems of border
Step 6 : The pen makes burrs but it is very small, I know.

**Comment by jmoenig** [comment]
Yep, that's an issue with the automatic anti-aliasing that happens in the HTML5 canvas element which we cannot (sensibly) turn off (and probably wouldn't wish to).

**Comment by brianharvey** [comment]
Would it be possible to maintain two canvases, the one we actually display and another one with greater precision and no anti-aliasing?  Then fill could use the latter to determine the border of the region.

**Comment by jmoenig** [comment]
the HTML Canvas API doesn't let you turn off antialiasing. People do it by resorting to dirty tricks and ugly kludges such as making sure never to draw on a sub-pixel that end up implementing their own primitives. No good.

**Comment by gdavid04** [comment]
maybe [https://stackoverflow.com/a/195575/6506122](https://stackoverflow.com/a/195575/6506122)
